workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Jobs/CodegenJob.cs

```diff
@@ -45,31 +81,48 @@ public void Clean()
                 var remainingFilesInFolder = fileSystem.GetFilesInDirectory(fileInfo.DirectoryPath);
                 if (remainingFilesInFolder.Count == 0)
                 {
+                    logger.Info($"Deleting output directory {fileInfo.DirectoryPath}");
                     fileSystem.DeleteDirectory(fileInfo.DirectoryPath);
                 }
             }
+
+            if (numRemovedDirectories > 0)
+            {
+                logger.Info($"Directories cleaned: {numRemovedDirectories}");
+            }
         }
 
         public void Run()
         {
+            var jobType = GetType();
+            logger.Info($"Starting {jobType}");
+
             RunImpl();
 
+            logger.Info("Writing generated code to disk");
             foreach (var entry in Content)
             {
                 var fileInfo = fileSystem.GetFileInfo(Path.Combine(OutputDirectory, entry.Key));
 
                 if (!fileSystem.DirectoryExists(fileInfo.DirectoryPath))
                 {
+                    logger.Trace($"Creating output directory {fileInfo.DirectoryPath}");
                     fileSystem.CreateDirectory(fileInfo.DirectoryPath);
                 }
 
+                logger.Trace("Fixing line endings");
                 // Fix up line endings
                 var contents = entry.Value
                     .Replace("\r\n", "\n")
                     .Replace("\n", Environment.NewLine);
 
                 fileSystem.WriteToFile(fileInfo.CompletePath, contents);
+                logger.Trace($"Written {fileInfo.CompletePath}");
```

Comment:
Arguably should be `Info`

Reply:
I'd argue against that, as it would just be clutter and (imo) not essential unless you're trying to debug an issue